include/sys/dsl_dataset.h

```diff
@@ -188,6 +191,9 @@ typedef struct dsl_dataset {
 	 */
 	refcount_t ds_longholds;
 
+	/* refcount on our encryption key mapping */
+	uint32_t ds_key_mappings;
```

Comment:
Its not really. This used to be more important to the implementation than it is. I can replace it.